test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -443,6 +444,11 @@ static int test_overall_capabilities(void)
 				return -1;
 			}
 		}
+
```

Comment:
Checking of max_schedulers_per_node capability is not added by this patch or by any other patch, even though the TM configuration used in the validation tests seems to require that max_schedulers_per_node is at least NUM_PRIORITIES (which is now 4).

Maybe it would be good to separate the addition of the max_schedulers_per_node capability into its own API patch and its own implementation and validation patches. Then the API change would not be hidden in an unrelated commit and it would be obvious that validation test changes are missing.

Reply:
Added it in v7 to check if max_schedulers_per_node  <= ODP_TM_MAX_PRIORITIES.

Reply:
The reason, I didn't add check for it to be NUM_PRIORITIES is because, in packet priority mode overwrite, since current tests are having node priority 0, it works in platforms that support even max_schedulers per node = 1.

Reply:
> Added it in v7 to check if max_schedulers_per_node <= ODP_TM_MAX_PRIORITIES.

I do not see why we want to check that max_scheduelrs_per_node capability is *less* than some value. My point is that the tests require certain number of schedulers per node and should check the capability against that number.

> it works in platforms that support even max_schedulers per node = 1.

In some platforms max_schedulers_per_node could be zero. So you should check that the capability is at least one before configuring the TM topology (the API says that the TM config must be such that the scheduler limit is not exceeded, even if you are not going to send single packet through the TM system).

TM priority and topology configuration in the test setup is such that NUM_QUEUES_PER_NODE (which now equals NUM_PRIORITIES) queues of distinct priorities are connected to the level 2 nodes and multiple level 2 node connect to the same level 1 node. Thus in the "preserve" packet priority mode the level 1 nodes need NUM_QUEUES_PER_NODE schedulers. Creating such a config is against the API rules if max_schedulers_per_node is not at least NUM_QUEUES_PER_NODE.